test/ldap_test.rb

```diff
@@ -114,6 +115,23 @@ def test_search_strategy_misconfigured_to_unrecognized_strategy_falls_back_to_de
     assert_equal GitHub::Ldap::MemberSearch::Recursive,         @ldap.member_search_strategy
   end
 
+  def test_user_search_strategy_active_directory_when_configured_and_has_capability
+    @ldap.stubs(:active_directory_capability?).returns(true)
+    @ldap.configure_user_search_strategy true
+    assert_equal GitHub::Ldap::UserSearch::ActiveDirectory, @ldap.user_search_strategy.class
+  end
+
+  def test_user_search_strategy_not_active_directory_without_capability
+    @ldap.stubs(:active_directory_capability?).returns(false)
+    @ldap.configure_user_search_strategy true
+    assert_equal GitHub::Ldap::UserSearch::Default, @ldap.user_search_strategy.class
```

Comment:
@davesims the approach we've taken in the past is to accept configuration as dictate and not try to be too smart about it; that is, if you specify you want some specific behavior, it would be surprising (in a bad way) to not get that behavior, even if you've configured unsupported behavior.